experimental/skills/DialogToDialog/DialogRootBot/SkillsConfiguration.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.BotBuilderSamples.DialogRootBot
+{
+    /// <summary>
+    /// A helper class that loads Skills information from configuration.
+    /// </summary>
+    public class SkillsConfiguration
+    {
+        public SkillsConfiguration(IConfiguration configuration)
+        {
+            var section = configuration?.GetSection("BotFrameworkSkills");
+            var skills = section?.Get<BotFrameworkSkill[]>();
+            if (skills != null)
+            {
+                foreach (var skill in skills)
+                {
+                    Skills.Add(skill.Id, skill);
+                }
+            }
+
+            var skillHostEndpoint = configuration?.GetValue<string>(nameof(SkillHostEndpoint));
+            if (!string.IsNullOrWhiteSpace(skillHostEndpoint))
```

Comment:
Should it generate an exception any time the endpoint is not a valid URI?

Reply:
That validation is done by the new Uri() constructor, so don't need to add it here. Just a helper in case the forget to set it. 

---
In reply to: [356242229](https://github.com/microsoft/BotBuilder-Samples/pull/2032#discussion_r356242229) [](ancestors = 356242229)